src/qs8-packw/gen/qs8-packw-x8c4-gemm-goi-avxvnni-prfm.c

```diff
@@ -0,0 +1,450 @@
+        const __m256i v2_01234567 = _mm256_loadu_si256((const __m256i*) w2);
+        const __m256i v3_01234567 = _mm256_loadu_si256((const __m256i*) w3);
+        const __m256i v4_01234567 = _mm256_loadu_si256((const __m256i*) w4);
+        const __m256i v5_01234567 = _mm256_loadu_si256((const __m256i*) w5);
+        const __m256i v6_01234567 = _mm256_loadu_si256((const __m256i*) w6);
+        const __m256i v7_01234567 = _mm256_loadu_si256((const __m256i*) w7);
+
+        const __m256i v01_0145 = _mm256_unpacklo_epi32(v0_01234567, v1_01234567);
+        const __m256i v01_2367 = _mm256_unpackhi_epi32(v0_01234567, v1_01234567);
+        const __m256i v23_0145 = _mm256_unpacklo_epi32(v2_01234567, v3_01234567);
+        const __m256i v23_2367 = _mm256_unpackhi_epi32(v2_01234567, v3_01234567);
+        const __m256i v45_0145 = _mm256_unpacklo_epi32(v4_01234567, v5_01234567);
+        const __m256i v45_2367 = _mm256_unpackhi_epi32(v4_01234567, v5_01234567);
+        const __m256i v67_0145 = _mm256_unpacklo_epi32(v6_01234567, v7_01234567);
+        const __m256i v67_2367 = _mm256_unpackhi_epi32(v6_01234567, v7_01234567);
+
+        const __m256i v02_02 = _mm256_unpacklo_epi64(v01_0145, v23_0145);
+        const __m256i v02_13 = _mm256_unpackhi_epi64(v01_0145, v23_0145);
+        const __m256i v13_02 = _mm256_unpacklo_epi64(v01_2367, v23_2367);
+        const __m256i v13_13 = _mm256_unpackhi_epi64(v01_2367, v23_2367);
+        const __m256i v46_02 = _mm256_unpacklo_epi64(v45_0145, v67_0145);
+        const __m256i v46_13 = _mm256_unpackhi_epi64(v45_0145, v67_0145);
+        const __m256i v57_02 = _mm256_unpacklo_epi64(v45_2367, v67_2367);
+        const __m256i v57_13 = _mm256_unpackhi_epi64(v45_2367, v67_2367);
+
+       const __m256i v04_0 = _mm256_permute2f128_si256(v02_02, v46_02, _MM_SHUFFLE(0, 2, 0, 0));
+       const __m256i v04_1 = _mm256_permute2f128_si256(v02_02, v46_02, _MM_SHUFFLE(0, 3, 0, 1));
+       const __m256i v15_0 = _mm256_permute2f128_si256(v02_13, v46_13, _MM_SHUFFLE(0, 2, 0, 0));
+       const __m256i v15_1 = _mm256_permute2f128_si256(v02_13, v46_13, _MM_SHUFFLE(0, 3, 0, 1));
+       const __m256i v26_0 = _mm256_permute2f128_si256(v13_02, v57_02, _MM_SHUFFLE(0, 2, 0, 0));
+       const __m256i v26_1 = _mm256_permute2f128_si256(v13_02, v57_02, _MM_SHUFFLE(0, 3, 0, 1));
+       const __m256i v37_0 = _mm256_permute2f128_si256(v13_13, v57_13, _MM_SHUFFLE(0, 2, 0, 0));
+       const __m256i v37_1 = _mm256_permute2f128_si256(v13_13, v57_13, _MM_SHUFFLE(0, 3, 0, 1));
+
+        xnn_prefetch_to_l1((const int8_t*) w0 + 448);
+        xnn_prefetch_to_l1((const int8_t*) w1 + 448);
+        xnn_prefetch_to_l1((const int8_t*) w2 + 448);
+        xnn_prefetch_to_l1((const int8_t*) w3 + 448);
+        xnn_prefetch_to_l1((const int8_t*) w4 + 448);
+        xnn_prefetch_to_l1((const int8_t*) w5 + 448);
+        xnn_prefetch_to_l1((const int8_t*) w6 + 448);
+        xnn_prefetch_to_l1((const int8_t*) w7 + 448);
```

Comment:
if you have time, try some different offsets here and run on meteorlake, alderlake or lunar lake.  
448 is 7 cache lines of 64 and a good choice
960 is 15 cache lines of 64 and may work better on large packw
128 is 2 cache lines ahead and sometimes good when doing a small amount of loads.  You are reading 32 bytes, so these prefetches are somewhat redundent.
4096 is interesting due to page

instead of prefetching the same cachelines, it might be worth the logic to only prefetch half the time

Reply:
I have tried those different offsets on MTL on benchmark qs8_packw_bench, and 960 has the lowest cache miss, but 448 has better performance. for alternating prefetch, instructions is reduced, but totally there is no performance improvement.